hybrid/compartment_examples/inter_comp_call_secure/main.c

```diff
@@ -0,0 +1,194 @@
+/* In this example, we tighten security even further. For each compartment we
+ * create, we store some capabilities of interest. Essentially, we setup
+ * certain entry points, and make them available to each compartment.
+ * Currently, these involve the PCC/DDC of the `switch_compartment` function
+ * (PCC to execute, DDC to access memory where we store information pertaining
+ * to compartments; essentially, `switch_compartment` performs a limited
+ * privelege escalation, but only over prepared data within `main`).
+ * Additionally, we also seal the PCC of `switch_compartment`.
+ *
+ * Note that in the current design, each compartment has a single function it
+ * executes when it is called. Allowing multiple entry points can be modelled
+ * by having more capabilities be saved, in addition to the compartment
+ * switching ones, in each compartment's heap. Alternatively, we could, within
+ * the compartment switcher, determine in which compartment we need to switch
+ * to call a given function, based on PCC bounds for each compartment.
+
+ * Additionally, we also changed the expected memory layout in the
+ * compartments. In the absence of a custom memory allocator, we note that the
+ * heap is not used, except for manually storing capabilities as detailed
+ * above. Current compartment memory layout:
+ *
+ * `DDC value` > -----------------
+ *              |                 |
+ *              |     HEAP        |
+ *              |                 |
+ *              |   saved caps    |
```

Comment:
What is included in "saved caps"? I guess these are the "some capabilities of interest" mentioned earlier, but I must admit that I don't know precisely what that set might be!

Reply:
This is detailed in `main.c:4`:
> Currently, these involve the PCC/DDC of the `switch_compartment` function (PCC to execute, DDC to access memory where we store information pertaining to compartments

Further, there is a `saved_caps` variable a bit down. I can add in an underscore to make the connection more obvious?